rest/main_legacy_test.go

```diff
@@ -0,0 +1,68 @@
+package rest
+
+import (
+	"flag"
+	"testing"
+
+	"github.com/couchbase/sync_gateway/base"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// Test legacy flags using valid values
+func TestLegacyFlagsValid(t *testing.T) {
```

Comment:
Is it possible to write a test that ensures legacy flags do not override non-legacy flags?
Especially around the special handling for `-verbose`/`-logging.console.log_level` and "normal" handling for something like `-url`/`-bootstrap.server`